engine/alternative-runtimes.md

```diff
@@ -0,0 +1,285 @@
+
+## How to use alternative runtimes
+
+There are different ways you can go about selecting an alternative container
+runtime.
+
+- [Configure the default runtime](#configure-the-default-runtime)
+- [Register additional runtimes](#register-additional-runtimes)
+- [Use containerd shims](#use-containerd-shims)
+
+### Configure the default runtime
+
+You can change the default runtime of the daemon, using the `default-runtime`
+option.
+
+The following examples show how to configure the daemon to use gVisor (`runsc`)
+as the default runtime:
+
+- Using the daemon configuration file (`/etc/docker/daemon.json`)
+
+  ```json
+  {
+    "default-runtime": "/usr/bin/runsc"
+  }
+  ```
+
+- Using the `dockerd` CLI
+
+  ```console
+  $ dockerd --default-runtime /usr/bin/runsc
+  ```
+
+### Register additional runtimes
+
+You can register additional runtimes with the Docker daemon on startup.
+This doesn't change the default runtime, but makes it possible to select
+alternative runtimes on a per-container basis.
+
+The following examples show how to register `youki` as a container runtime:
+
+- Using the daemon configuration file (`/etc/docker/daemon.json`)
+
+  ```json
+  {
+    "runtimes": {
+      "youki": {
+        "path": "/usr/bin/youki",
+        "runtime-args": ["--debug"]
+      }
+    }
+  }
+  ```
+
+- Using the `dockerd` CLI
+
+  ```console
+  $ dockerd --add-runtime /usr/bin/youki
+  ```
+
+> **Note**
+>
+> You can specify additional arguments when you register a runtime using the
+> configuration file, but not when you use the CLI flag.
+
+### Use containerd shims
+
+Docker Engine supports the use of containerd supports runtime shims.
+Using shims, you can use alternative runtimes without having to configure or
+restart the Docker daemon.
+
+To use a containerd shim, you must install the shim binary on `PATH` on the
+system where the Docker daemon is running. With the shim installed,
+you can select the shim with `docker run` using the shim's fully qualified name.
+Fully qualified shim names take the form of `io.containerd.<shim>.<version>`.
+
+The following `docker run` command uses the Kata Containers runtime, provided
+that the shim binary is installed on `PATH`.
+
+```console
+$ docker run --runtime io.containerd.kata.v2 hello-world
+```
+
+#### Configure shims
+
+If you need to pass additional configuration for a containerd shim, you can
+use the `runtimes` option in the daemon configuration file.
+
+1. Edit the daemon configuration file by adding a `runtimes` entry for the
+   shim you want to configure.
+
+   - Specify the fully qualified name for the runtime in `runtimeType` key
+   - Add your runtime configuration under the `options` key
+
+   ```json
+   {
+     "runtimes": {
+       "runsc": {
+         "runtimeType": "io.containerd.runsc.v1",
+         "options": {
+           "file-access": "shared"
+         }
+       }
+     }
+   }
+   ```
+
+2. Restart the Docker daemon.
+
+   ```console
+   # systemctl restart docker
```

Comment:
using the pound char as prompt to denote root user (here and throughout the doc) but maybe using sudo is better?

Reply:
The pound-char prompt denoting root shell is a documentation convention I am very familiar with, at least.